File: packages/java/parser-jvm-plugin-backbone/src/test/java/dev/hilla/parser/plugins/backbone/complextype/ComplexTypeEndpoint.java
package dev.hilla.parser.plugins.backbone.complextype;

import java.util.List;
import java.util.Map;

@Endpoint
public class ComplexTypeEndpoint {
    public ComplexTypeModel getComplexTypeModel(
            List<Map<String, String>> data) {
        return new ComplexTypeModel();
    }

    private ComplexTypeModel getPrivateComplexTypeModel() {
        return new ComplexTypeModel();
    }

    public static class ComplexTypeModel {
        private List<Map<String, List<String>>> complexList;
        private Map<String, List<String>> complexMap;
    }
}

File: packages/java/parser-jvm-plugin-backbone/src/test/resources/com/vaadin/hilla/parser/plugins/backbone/complextype/expected/ComplexTypeEndpoint.ts
import { EndpointRequestInit as EndpointRequestInit_1 } from "@vaadin/hilla-frontend";
import type ComplexTypeModel_1 from "./com/vaadin/hilla/parser/plugins/backbone/complextype/ComplexTypeEndpoint/ComplexTypeModel.js";
import client_1 from "./connect-client.default.js";
async function getComplexTypeModel_1(data: Array<Record<string, string | undefined> | undefined> | undefined, init?: EndpointRequestInit_1): Promise<ComplexTypeModel_1 | undefined> { return client_1.call("ComplexTypeEndpoint", "getComplexTypeModel", { data }, init); }
export { getComplexTypeModel_1 as getComplexTypeModel };
